### ISSUE TITLE
Code logic

### ISSUE THREAD

**Issue by HEasoner** [Shelton1013/Chain_of_Attack#1]
embedding_sim1 = torch.mean(torch.sum(cur_fused_embedding * cle_fused_embedding, dim=1)) # pos
            embedding_sim2 = torch.mean(torch.sum(cur_fused_embedding * tgt_fused_embedding, dim=1)) # neg
            
            """
            # triplet loss, origin triplet loss use l2 dis, we use cos sim
            # anchor = cur_fused_embedding
            # positive = tgt_fused_embedding
            # negative = cle_fused_embedding
            # loss = triplet_loss(anchor, positive, negative)
            """

            margin = 1 - p_neg
            loss = torch.mean(torch.relu(embedding_sim2 - p_neg * embedding_sim1 + margin))
            loss.backward()

Great Work!
I have a question. The loss here is to make sim2 as small as possible and sim1 as large as possible, but we want the current embedding to be closer to the tgt so sim2 as large as possible. Or I have a problem with my understanding?
